Added points functionality.

patrycja93 opened the pull request:
The Judge class displays players points and add points to players after each match.

patrycja93 committed:
Added points functionality. The Judge class displays players points and add points to players after each match.

BulaMar commented on src/main/java/pl/patrycja/ox/ScoreBoard.java:
```diff
@@ -0,0 +1,31 @@
+        this.playerList = playerList;
+    }
+
+    public void addWinnerPoints(Player player, int howMuch) {
```
Will points for wining match can differ or there also should be constant for those?

patrycja93 replied:
Ok I will correct it.

patrycja93 committed:
Updated tests.

BulaMar commented on src/main/java/pl/patrycja/ox/gameflow/Match.java:
```diff
@@ -37,7 +37,7 @@ public Match addPlayers(List<Player> players, Player activePlayer) {
     void start(int numberOfMatch) {
         //TODO: ask about language
-        board.startMatch(spectators, numberOfMatch, activePlayer);
+        board.startMatch(numberOfMatch, activePlayer);
```
🤔  interesting, shouldn't match be started?

BulaMar commented on src/main/java/pl/patrycja/ox/gameflow/Mode.java:
```diff
@@ -22,17 +23,19 @@ public Mode(UI ui) {
     void play(List<Player> players) {
         Player initialPlayer = askWhichPlayerStarts(players);
-
+        Match match = Match.init(gameSettings.getBoardSize(), ui, spectatorsList);;
```
This can be back in for loop as it was.

BulaMar commented on src/main/java/pl/patrycja/ox/winnerchecker/Judge.java:
```diff
@@ -11,35 +12,32 @@
 class Judge implements Spectator {
 
-    private List<WinnerChecker> winnerCheckers;
+    private static final int POINTS_FOR_WIN = 3;
```
Move it to score board.

BulaMar commented on src/main/java/pl/patrycja/ox/winnerchecker/JudgeStatements.java:
```diff
@@ -0,0 +1,61 @@
+public class JudgeStatements {
+
+    String showGameSummary(List<Player> players) {
+        final int maxAmountOfPoints = getMaxAmountOfPoints(players);
```
maybe `getHighScore?`

BulaMar commented on src/main/java/pl/patrycja/ox/winnerchecker/JudgeStatements.java:
```diff
@@ -0,0 +1,61 @@
+        final int maxAmountOfPoints = getMaxAmountOfPoints(players);
+
+        List<Player> winners = checkWinners(players, maxAmountOfPoints);
+        List<Player> losers = checkLosers(players, maxAmountOfPoints);
```
you do not need them when is draw, maybe it will be better to check it first?

BulaMar reviewed (changes requested):
Each time you are doing better, but still some things are missing.

patrycja93 committed:
After code review.
Added new Enum for helping with returns information about put sign to board. Little changes in UI. Corrected automatic tests. Added Input checker to UI. Changed Judge class and Assessor class to
implements observator pattern.

patrycja93 commented:
Little changes in classes according to your tips.

BulaMar commented on src/main/java/pl/patrycja/ox/GameSettings.java:
```diff
@@ -94,7 +79,7 @@ public int getUnbrokenLine() {
 
     /**
-     * Return match number
+     * Return number of match
```
matches

BulaMar commented on src/main/java/pl/patrycja/ox/Observable.java:
```diff
@@ -2,18 +2,15 @@
-     */
 
-    void startMatch(List<Spectator> spectators, int number, Player player);
+    void notifySpectators(int field, Player player);
```
missing documentation

BulaMar commented on src/main/java/pl/patrycja/ox/Observable.java:
```diff
@@ -2,18 +2,15 @@
-    void startMatch(List<Spectator> spectators, int number, Player player);
+    void notifySpectators(int field, Player player);
+
+    void subscribe(Spectator spectator);
```
missing documentation

BulaMar commented on src/main/java/pl/patrycja/ox/gameflow/TicTacToe.java:
```diff
@@ -20,11 +20,9 @@
     public static void main(String[] args) {
 
-        ModeFactory modeFactory = new ModeFactory(getLessThanFour(args));
+        ModeFactory modeFactory = new ModeFactory(getLessThanFour(args), args);
```
Do you need both?

BulaMar commented on src/main/java/pl/patrycja/ox/ui/ConsoleUI.java:
```diff
@@ -4,18 +4,44 @@
 
 class ConsoleUI implements UI {
 
+    public static final String ANSI_YELLOW = "\u001B[33m";
```
🥇

BulaMar commented on src/main/java/pl/patrycja/ox/ui/ConsoleUI.java:
```diff
@@ -4,18 +4,44 @@
+            displayError("Entered wrong arguments. Please run the game again with correct integer numbers.");
+        }
+        if (inputChecker.checkIfUnbrokenLineIsGraterThanBoardSize(args)) {
+            displayError("Unbroken number of sign cannot be greater then board size and less than 3. Values was changed.\n");
```
signs

BulaMar reviewed (approved):
LGTM 👍  only minor things

patrycja93 committed:
Little changes after code review and automated test corrections
